src/ResourceManagement/Network/NicIPConfigurationImpl.cs

```diff
@@ -166,89 +167,82 @@ internal NicIPConfigurationImpl WithNewPublicIPAddress()
         {
             removePrimaryPublicIPAssociation = true;
             return this;
         }
 
-        
+
         ///GENMHASH:0FBBECB150CBC82F165D8BA614AB135A:055A43C17C98D1AD747257B0659283D6
         internal NicIPConfigurationImpl WithSubnet(string name)
         {
             subnetToAssociate = name;
             return this;
         }
 
-        
-        ///GENMHASH:D36B69B83A3C752672806F0242C22209:FBE8ED51BA892F18B9FBCE3042C8244C
+
+        ///GENMHASH:D36B69B83A3C752672806F0242C22209:760CFCD09BEBB96F67F6C9E6256E6656
         internal NicIPConfigurationImpl WithExistingLoadBalancerBackend(ILoadBalancer loadBalancer, string backendName)
         {
-            foreach (var pool in loadBalancer.Inner.BackendAddressPools)
+            if (loadBalancer != null)
             {
-                if (pool.Name.Equals(backendName, StringComparison.OrdinalIgnoreCase))
+                foreach (var pool in loadBalancer.Inner.BackendAddressPools)
                 {
-                    EnsureBackendAddressPools().Add(pool);
-                    return this;
+                    if (pool.Name.Equals(backendName, StringComparison.OrdinalIgnoreCase))
+                    {
+                        EnsureLoadBalancerBackendAddressPools().Add(pool);
+                        return this;
+                    }
                 }
             }
 
-            return this;
+            return null;
         }
 
-        
-        ///GENMHASH:03CBA85933E5B90121E4F4AE70F457EE:95BCC3AF459B4DC5CF3EC334E7D52AE2
+
+        ///GENMHASH:03CBA85933E5B90121E4F4AE70F457EE:E61FBBEB12B90725D7D58EC6BCF1454D
         internal NicIPConfigurationImpl WithExistingLoadBalancerInboundNatRule(ILoadBalancer loadBalancer, string inboundNatRuleName)
         {
-            foreach (var rule in loadBalancer.Inner.InboundNatRules)
+            if (loadBalancer != null)
             {
-                if (rule.Name.Equals(inboundNatRuleName, StringComparison.OrdinalIgnoreCase))
+                foreach (var rule in loadBalancer.Inner.InboundNatRules)
                 {
-                    EnsureInboundNatRules().Add(rule);
-                    return this;
+                    if (rule.Name.Equals(inboundNatRuleName, StringComparison.OrdinalIgnoreCase))
+                    {
+                        EnsureInboundNatRules().Add(rule);
+                        return this;
+                    }
                 }
-            }
 
-            return this;
-        }
-
-        
-        ///GENMHASH:63C8240EDA17974D47E176085F22436A:2D595A527D306417A72C1C52A5906DB3
-        private IList<BackendAddressPoolInner> EnsureBackendAddressPools()
-        {
-            var poolRefs = Inner.LoadBalancerBackendAddressPools;
-            if (poolRefs == null)
-            {
-                poolRefs = new List<BackendAddressPoolInner>();
-                Inner.LoadBalancerBackendAddressPools = poolRefs;
             }
 
-            return poolRefs;
+            return null;
```

Comment:
>return null; [](start = 12, length = 12)

here as well. Shouldn't you return this not to break fluent flow?

Reply:
actually, same thing here - the flow shouldn't really proceed because it is assuming a child that doesn't exist. 
So I used Java's semantic for missing Map elements: return null rather than throw exception.
(Although .NET Dictionary doesn't quite work that way (unfortunately), HybridDictionary does, so that precedent made me feel it's ok on the .NET side to do the same.Leave a comment

Reply:
Please throw InvalidOperationException instead.

Reply:
ok